### PR TITLE
feat(curve-metapool): added pools and interfaces for Curve metapool's…

### PR DESCRIPTION
**Pull request by mariogutval**
Curve metapools adapters

Added necessary interfaces and pools for CurveMetapoolDepositAdapter and CurveMetapoolSwapAdapter.

### CODE REVIEW

**Review by dhruvinparikh — CHANGES_REQUESTED** [review]
- [x] Please have uniformity in names. `ABC.json` should have corresponding `ABC.sol`. `aBC.sol` or `aBc.sol` is not fine.
- [ ] Please include pool data in contracts.ts
- [ ] Please have metapool factory address and its ABI in `contracts.ts`
- [ ] Please include metapool data in `contracts.ts`

File: src/ethereum/curve/contracts.ts [hunk 1]
```diff
@@ -5,6 +5,9 @@ import poolTokenV2Abi from "./abi/CurvePoolTokenV2.json";
 import poolTokenV3Abi from "./abi/CurvePoolTokenV3.json";
 import zapAbi from "./abi/CurveZap.json";
 import metaZapAbi from "./abi/CurveMetaZap.json";
+import metapoolFactory from "./abi/CurveMetapoolFactory.json";
```
Comment (deepcrazy) [inline src/ethereum/curve/contracts.ts]: Can we have format? like this:
```suggestion
import CurveMetapoolFactoryAbi from "./abi/CurveMetapoolFactory.json";
```

to maintain uniformity with lines 14-21.

cc: @dhruvinparikh

**Review by deepcrazy — CHANGES_REQUESTED** [review]
Just a small doubt about imports in the `contract.ts` file.

File: src/ethereum/curve/contracts.ts (hunk 1 above)
Comment (mariogutval) [reply src/ethereum/curve/contracts.ts]: We should format previous lines as well, right?

Comment (NouDaimon) [inline src/ethereum/curve/contracts.ts]: All of the imports should be `CamelCased` and matching their JSON filenames:

`import CurvePoolTokenV3Abi from "./abi/CurvePoolTokenV3.json;`

File: src/ethereum/curve/contracts.ts [hunk 2]
```diff
@@ -27,6 +30,9 @@ const contracts = {
   metaSwapAbi,
   zapAbi,
   metaZapAbi,
+  metapoolFactory,
```
Comment (NouDaimon) [inline src/ethereum/curve/contracts.ts]: All Abis should follow this format: 

`CurveMetaPoolDeposit: { abi: CurveMetaPoolDepositAbi } ` 

If an address is needed:

`CurveMetaPoolDeposit: { 
        abi: CurveMetaPoolDepositAbi,
        address: "someAddress"
 } `

Comment (NouDaimon) [reply src/ethereum/curve/contracts.ts]: With the correct indentation on the address field!

**Review by NouDaimon — CHANGES_REQUESTED** [review]

File: src/ethereum/curve/contracts.ts [hunk 3]
```diff
@@ -1077,6 +1083,4780 @@ const contracts = {
       tokens: ["0x3Ed3B47Dd13EC9a98b44e6204A523E766B225811"],
     },
   },
+  CurveMetapoolDepositPool: {
```
Comment (NouDaimon) [inline src/ethereum/curve/contracts.ts:1832]: If there are pools for deposits and swaps, please split them as such. I am unaware what pools we need for Curve, however all of them should be captured in this file with the appropriate naming for identification.

**Review by NouDaimon — CHANGES_REQUESTED** [review]

File: src/ethereum/curve/contracts.ts (hunk 1 above)
Comment (deepcrazy) [reply src/ethereum/curve/contracts.ts]: Yes, right..

**Review by NouDaimon — APPROVED** [review]

**Review by dhruvinparikh — APPROVED** [review]

**Pull request merged**